internal/decoder/stream.go

```diff
@@ -344,6 +345,7 @@ func (s *Stream) skipArray(depth int64) error {
 						s.cursor = cursor
 						if s.read() {
 							_, cursor, p = s.statForRetry()
```

Comment:
what about add new parameter in `statForRetry` and pass parameters? (or make new function)

```
func (s *Stream) statForRetry(rewindCursor bool) ([]byte, int64, unsafe.Pointer) {
	if rewindCursor {
		s.cursor-- // for retry ( because caller may progress cursor position in each loop )
	}
	return s.buf, s.cursor, (*sliceHeader)(unsafe.Pointer(&s.buf)).data
}
```

It may confuse when additional cursor increments happens outside of `statForRetry` function